src/dolphin/unwrap/_unwrap.py

```diff
@@ -272,12 +286,47 @@ def unwrap(
             output_filename=combined_mask_file,
         )
 
+    if run_goldstein:
+        suf = Path(unw_filename).suffix
+        if suf == ".tif":
+            driver = "GTiff"
+            opts = list(io.DEFAULT_TIFF_OPTIONS)
+        else:
+            driver = "ENVI"
+            opts = list(io.DEFAULT_ENVI_OPTIONS)
```

Comment:
Actually I also meant to comment here- `get_raster_driver` is a good one to add [to the other similar io functions](https://github.com/isce-framework/dolphin/blob/main/src/dolphin/io/_core.py#L34-L42), but here I might actually move to cut even the ENVI one and just produce geotiff outputs at this stage. I think Ryan got this ENVI check back when I was half way out of only setting up ENVI files before the wrapper/tophu was set up.

If we do want to accomodate more than GTiff, that would probably make more sense as a larger refactor to pass around `DatasetWriters`. It would just require more thought how you would swap the user's `DatasetWriter` for a temporary one to hold the goldstein output